.github/workflows/native-images.yml

```diff
@@ -0,0 +1,115 @@
+name: Build Native Images
+defaults:
+  run:
+    shell: bash
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        type: string
+        description: 'Release version (e.g. v1.2.3); omit to skip attaching artifacts to a release'
+        required: false
+      release_repo:
+        type: string
+        description: 'The repository hosting the github release; used only if the release version parameter is set'
+        required: false
+        default: temporalio/sdk-java
+      clobber:
+        type: boolean
+        description: 'Overwrite existing release artifacts'
+        required: false
+        default: no
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      matrix:
+        include:
+          - dist: ubuntu-latest
+            os_family: linux
+            arch: amd64
+          - dist: macos-latest
+            os_family: macOS
+            arch: amd64
+          - dist: windows-latest
+            os_family: windows
+            arch: amd64
+    runs-on: ${{ matrix.dist }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+        
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        
+      - name: Build Native Image
+        run: ./gradlew :temporal-test-server:build
+      
+      # path ends in a wildcard because on windows the file ends in '.exe'
+      # path excludes *.txt because native-image also writes a build manifest txt file
+      - name: Upload executable to workflow
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.os_family }}_${{ matrix.arch }}
+          path: |
+            temporal-test-server/build/graal/temporal-test-server*
+            !temporal-test-server/build/graal/*.txt
+          if-no-files-found: error
+          retention-days: 1
+          
+  attach_to_release:
+    name: Attach to release
+    needs: build
+    if: github.event.inputs.release
+    runs-on: ubuntu-latest
+    steps:
+    
+      # when no artifact is specified, all artifacts are downloaded and expanded into CWD
+      - name: Fetch executables
+        uses: actions/download-artifact@v2
+        
+      # example: linux_amd64/ -> temporal-test-server_1.2.3_linux_amd64
+      # the name of the directory created becomes the basename of the archive (*.tar.gz or *.zip) and 
+      # the root directory of the contents of the archive.
+      - name: Rename dirs
+        run: |
+          shopt -s nullglob
+          version="$(sed 's/v//'<<<'${{ github.event.inputs.release }}')"
+          for dir in *; do
+            mv "$dir" "temporal-test-server_$version_$dir"
+          done        
```

Comment:
These scripts are short but IMO we should be putting all of them in external files rather than inline. Just dramatically easier to work with in IDEs etc

Also PR 1k :tada:

Reply:
I agree, I'd ideally want scripts that are more than a line long to not be shell scripts but I won't block the PR on it.
We should also verify that github sets `-euo pipefail` for these scripts if we plan to leave them here.

Reply:
I did check - bash defaults to  `set -eo pipefail` but not with `-u`. I can add `-u` in defaults.run.bash at the top of this file (and `-O nullglob` for that matter)

Reply:
I can convert these stanzas to files but there's one good and one half-good reason not to do so - LMK if this changes your opinion...

The good reason is that as inline snippets, GH will include the code verbatim in the output of the action such that when it fails you get the code, the output, and the error in a little web UI that shows all the jobs and steps. If the code is in a separate file you just see the filename rather than the code itself. Debugging build systems tends to be impoverished and I'd lean toward making as much information as possible available wherever the error is indicated.

The half-good reason is that these snippets are not really bash. The `${{ foo }}` syntax is effectively a macro language (including its own expression language) that gets expanded prior to execution by the GH action runner so something like shellcheck is going to get false positives. Even syntax coloring gets confused.

Reply:
I get it, technically you can have a 1 liner in the yml command that calls a script with provided variables and for the script itself you can `set -x` to show which line is executed but it's not critical.

PR `0x3e8`! 🎉